test/entry.spec.js

```diff
@@ -187,21 +235,109 @@ Object.keys(testAPIs).forEach((IPFS) => {
+        const payload2 = 'hello again'
+        const entry1 = await Entry.create(ipfs, testIdentity, 'A', payload1, [])
+        const entry2 = await Entry.create(ipfs, testIdentity, 'A', payload2, [entry1])
+        const entry2Multihash = await Entry.toMultihash(ipfs, entry2)
+        const final = await Entry.fromCID(ipfs, entry2Multihash)
+
+        assert.strictEqual(final.id, 'A')
+        assert.strictEqual(final.payload, payload2)
+        assert.strictEqual(final.next.length, 1)
+        assert.strictEqual(final.next[0], entry1.cid)
+        assert.strictEqual(final.v, 0)
+        assert.strictEqual(final.cid, expectedCid)
+      })
+
+      it('creates a entry from ipfs multihash of v0 entries', async () => {
+        const expectedCid = 'QmTLLKuNVXC95rGcnrL1M3xKf4dWYuu3MeAM3LUh3YNDJ7'
+        const entry1Cid = await dagNode.write(ipfs, 'dag-pb', v0Entries.helloWorld)
+        const entry2Cid = await dagNode.write(ipfs, 'dag-pb', v0Entries.helloAgain)
+        const final = await Entry.fromCID(ipfs, entry2Cid)
+
+        assert.strictEqual(final.id, 'A')
+        assert.strictEqual(final.payload, v0Entries.helloAgain.payload)
+        assert.strictEqual(final.next.length, 1)
+        assert.strictEqual(final.next[0], v0Entries.helloAgain.next[0])
+        assert.strictEqual(final.next[0], entry1Cid)
+        assert.strictEqual(final.v, 0)
+        assert.strictEqual(final.cid, entry2Cid)
+        assert.strictEqual(final.cid, expectedCid)
+      })
+
+      it('throws an error if ipfs is not present', async () => {
+        let err
+        try {
+          await Entry.fromCID()
+        } catch (e) {
+          err = e
+        }
+        assert.strictEqual(err.message, 'Ipfs instance not defined')
+      })
+
+      it('throws an error if CID is undefined', async () => {
+        let err
+        try {
+          await Entry.fromCID(ipfs)
+        } catch (e) {
+          err = e
+        }
+        assert.strictEqual(err.message, 'Invalid CID: undefined')
+      })
+    })
+
+    describe('fromMultihash', () => {
+      it('creates a entry from ipfs multihash', async () => {
+        const expectedMultihash = 'QmRyrbupohhmSoXxQ56XthEiK6YEqK5ZN8SysJdo8GGdHq'
+        const payload1 = 'hello world'
+        const payload2 = 'hello again'
+        const entry1 = await Entry.create(ipfs, testIdentity, 'A', payload1, [])
+        const entry2 = await Entry.create(ipfs, testIdentity, 'A', payload2, [entry1])
+        const entry2Multihash = await Entry.toMultihash(ipfs, entry2)
+        const final = await Entry.fromMultihash(ipfs, entry2Multihash)
+
+        assert.strictEqual(final.id, 'A')
+        assert.strictEqual(final.payload, payload2)
+        assert.strictEqual(final.next.length, 1)
+        assert.strictEqual(final.next[0], entry1.cid)
+        assert.strictEqual(final.v, 0)
+        assert.strictEqual(final.cid, expectedMultihash)
+      })
+
+      it('creates a entry from ipfs multihash of v0 entries', async () => {
```

Comment:
Excellent, great to have a test for this! 👍

Reply:
Thanks!